staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/typeconverter.go

```diff
@@ -35,25 +37,59 @@ type TypeConverter interface {
 	TypedToObject(*typed.TypedValue) (runtime.Object, error)
 }
 
+type typeResolver interface {
+	Type(schema.GroupVersionKind) *typed.ParseableType
+}
+
+type mapTypeResolver map[schema.GroupVersionKind]*typed.ParseableType
+
+func (m mapTypeResolver) Type(k schema.GroupVersionKind) *typed.ParseableType {
+	return m[k]
+}
+
 type typeConverter struct {
-	parser *managedfields.GvkParser
+	parser typeResolver
 }
 
 var _ TypeConverter = &typeConverter{}
 
-// NewTypeConverter builds a TypeConverter from a proto.Models. This
+// NewTypeConverter builds a TypeConverter from a spec.Swagger. This
```

Comment:
Aren't we supposed to be able to build directly from an OpenAPIv3? Shouldn't we take the `map[string]*spec.Schema`?

Reply:
Or should we have two methods for convenience? (since there is the pointer dance we could avoid). Or provide a method to avoid that? Not sure yet.

Reply:
https://github.com/kubernetes/kubernetes/pull/115324 is where I intended to make that change to OpenAPIV3 - limiting this PR to just switching to using the TypeConverter.

Reply:
OK good point, thanks